JASP-R-Interface/jaspResults/src/jaspTable.cpp

```diff
@@ -860,22 +860,80 @@ void jaspTable::rectangularDataWithNamesToHtml(std::stringstream & out, std::vec
 	out << "\t</table>\n";
 }
 
-void jaspTable::addFootnote(Rcpp::RObject message, Rcpp::RObject symbol, Rcpp::RObject col_names, Rcpp::RObject row_names)
+Json::Value footnotes::tableFields::rowsToJSON() const
+{ 
+	return rows.size() == 0 ? Json::nullValue : jaspJson::SetJson_to_ArrayJson(rows); 
+}
+
+Json::Value footnotes::tableFields::colsToJSON() const
 {
-	if(message.isNULL())
-		Rf_error("One would expect a footnote to at least contain a message..");
+	return cols.size() == 0 ? Json::nullValue : jaspJson::SetJson_to_ArrayJson(cols); 
+}
 
-	std::vector<Json::Value> colNames = jaspJson::RcppVector_to_VectorJson(col_names, false);
-	std::vector<Json::Value> rowNames = jaspJson::RcppVector_to_VectorJson(row_names, false);
+Json::Value footnotes::convertToJSON() const
+{
+	Json::Value notes(Json::arrayValue);
+	for (auto & keyval : _data)
+	{
+		Json::Value note(Json::objectValue); 
+		tableFields const & fields = keyval.second;
+		
+		note["text"]	= keyval.first;
+		note["symbol"]	= fields.symbol;
+		note["rows"]	= fields.rowsToJSON();
+		note["cols"]	= fields.colsToJSON();
+		
+		notes.append(note);
+	}
+	return notes;
+}
 
-	Json::Value note(Json::objectValue);
+void footnotes::convertFromJSON_SetFields(Json::Value footnotes)
+{
+	if (!footnotes.isNull())
+	{
+		for(Json::Value::UInt i=0; i<footnotes.size(); i++)
+		{
+			const Json::Value &	footnote	= footnotes[i];
+			const std::string	text		= footnote["text"].asString();
+			
+			_data[text].symbol	= footnote["symbol"].asString();
+			_data[text].rows	= jaspJson::ArrayJson_to_SetJson(footnote["rows"]);
+			_data[text].cols	= jaspJson::ArrayJson_to_SetJson(footnote["cols"]);
+		}
+	}
+}
+
+void footnotes::insert(std::string text, std::string symbol, std::vector<Json::Value> colNames, std::vector<Json::Value> rowNames)
+{
+	if (_data[text].symbol.empty())
+		_data[text].symbol = symbol;
+	
+	for (int i=0; i<colNames.size(); i++)
+		_data[text].cols.insert(colNames[i]);
```

Comment:
`for (int i=0; i<colNames.size(); i++)
		_data[text].cols.insert(colNames[i]);`

kan volgens mij worden:
`_data[text].cols.insert(colNames.begin(), colNames.end());` en ook voor rowNames